cedar-policy-core/src/est.rs

```diff
@@ -243,11 +243,24 @@ impl Policy {
 }
 
 impl Clause {
+    fn filter_slots(e: ast::Expr, is_when: bool) -> Result<ast::Expr, FromJsonError> {
+        let slots: Vec<_> = e.slots().map(|s| *s).collect();
+        if let Some(slot) = slots.first() {
```

Comment:
I believe we can avoid collecting into a Vec here (and thus allocating memory).

```suggestion
        let slots = e.slots().copied();
        if let Some(slot) = slots.next() {
```

Reply:
It doesn't seem to work because `e` is still borrowed, which is moved in one of the if branches.

Reply:
What if 
```
let first_slot = e.slots().copied().next();
if let Some(slot) = first_slot {
```
Since the type of `first_slot` is now `Option<Slot>`, it doesn't keep `e` borrowed

Reply:
Right. Please see the latest commit.